pkg/webhook/patch.go

```diff
@@ -360,15 +379,31 @@ func addPrometheusConfigMap(pod *corev1.Pod, app *v1beta2.SparkApplication) []pa
 	name := config.GetPrometheusConfigMapName(app)
 	volumeName := name + "-vol"
 	mountPath := config.PrometheusConfigMapMountPath
+	port := config.DefaultPrometheusJavaAgentPort
+	if app.Spec.Monitoring.Prometheus.Port != nil {
+		port = *app.Spec.Monitoring.Prometheus.Port
+	}
+	protocol := config.DefaultPrometheusPortProtocol
+
 	patchOps = append(patchOps, addConfigMapVolume(pod, name, volumeName))
 	vmPatchOp := addConfigMapVolumeMount(pod, volumeName, mountPath)
-	if vmPatchOp == nil {
+	portPatchOp := addContainerPort(pod, port, protocol)
+	if vmPatchOp == nil || portPatchOp == nil {
```

Comment:
Please separate the two conditions out and handle them separately.

Reply:
Done.